sachem/Controllers/RechercheInscriptionController.cs

```diff
@@ -0,0 +1,164 @@
+                db.Entry(inscription).State = EntityState.Modified;
+                db.SaveChanges();
+
+                TempData["Success"] = "Inscription modifiée avec succès!";
+                return RedirectToAction("Details","RechercheInscription",new { id = id_Inscription });
+            }
+            TempData["Erreur"] = "Erreur lors de la modification de l'inscription. N'oubliez pas qu'il est impossible de modifier l'inscription des anciennes sessions.";
+            return RedirectToAction("Details", "RechercheInscription", new { id = id_Inscription });
+        }
+
+        [NonAction]
+        private void ListeSession(int Session = 0)
+        {
+            var lSessions = from session in db.Session select session;
+            var slSession = new List<SelectListItem>();
+            slSession.AddRange(new SelectList(lSessions, "id_Sess", "NomSession", Session));
+
+            ViewBag.Session = slSession;
+        }
+
+        [NonAction]
+        private void ListeTypeInscription(int TypeInscription = 0)
+        {
+            var lTypeInscription = from typeinscription in db.p_TypeInscription select typeinscription;
+            var slTypeInscription = new List<SelectListItem>();
+            slTypeInscription.AddRange(new SelectList(lTypeInscription, "id_TypeInscription", "TypeInscription", TypeInscription));
+
+            ViewBag.TypeInscription = slTypeInscription;
+        }
+
+        [NonAction]
+        private void ListeStatut(int Statut = 0)
+        {
+            var lStatut = from statut in db.p_StatutInscription where statut.id_Statut != BROUILLON select statut;
+            var slStatut = new List<SelectListItem>();
+            slStatut.AddRange(new SelectList(lStatut, "id_Statut", "Statut", Statut));
+
+            ViewBag.Statut = slStatut;
+        }
+
+        [NonAction]
+        private IEnumerable<Inscription> Rechercher()
+        {
+            var sess = 0;
+            var type = 0;
+            var statut = 0;
+
+            if (Request.RequestType == "GET" && Session["DernRechInsc"] != null && (string)Session["DernRechInscUrl"] == Request.Url?.LocalPath)
+            {
+                var anciennerech = Session["DernRechInsc"].ToString();
+                var tanciennerech = anciennerech.Split(';');
+
+                if (tanciennerech[0] != "")
+                {
+                    sess = int.Parse(tanciennerech[0]);
+                }
+                if (tanciennerech[1] != "")
+                {
+                    type = int.Parse(tanciennerech[1]);
+                }
+                if (tanciennerech[2] != "")
+                {
+                    statut = int.Parse(tanciennerech[2]);
+                }
+            }
+            else
+            {
+                if (!string.IsNullOrEmpty(Request.Form["Session"]))
+                    int.TryParse(Request.Form["Session"], out sess);
+                else if (Request.Form["Session"] == null)
```

Comment:
S'il n'est pas null ou vide dans le if précédent, il va toujours être null dans le if ici.
Less code is better code